iceoryx_posh/source/popo/building_blocks/condition_listener.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/popo/building_blocks/condition_listener.hpp"
+#include "iceoryx_utils/error_handling/error_handling.hpp"
+
+namespace iox
+{
+namespace popo
+{
+ConditionListener::ConditionListener(ConditionVariableData& condVarData) noexcept
+    : m_condVarDataPtr(&condVarData)
+{
+}
+
+void ConditionListener::resetSemaphore() noexcept
+{
+    // Count the semaphore down to zero
+    bool hasFatalError = false;
+    while (!hasFatalError
+           && getMembers()
+                  ->m_semaphore.tryWait()
+                  .or_else([&](posix::SemaphoreError) {
+                      errorHandler(Error::kPOPO__CONDITION_VARIABLE_WAITER_SEMAPHORE_CORRUPTED_IN_RESET,
+                                   nullptr,
+                                   ErrorLevel::FATAL);
+                      hasFatalError = true;
+                  })
+                  .value())
+    {
+    }
+}
+
+void ConditionListener::destroy() noexcept
+{
+    m_toBeDestroyed.store(true, std::memory_order_relaxed);
+    getMembers()->m_semaphore.post().or_else([](auto) {
```

Comment:
superfluous `(auto)`

Reply:
Sadly we require this parameter since `or_else` always provides the error value reference to the callable - event when we do not use it.